python/ray/tune/sample.py

```diff
@@ -6,9 +6,64 @@
 
 import numpy as np
 
+# Backwards compatibility
+try:
+    # Added in numpy>=1.17 but we require numpy>=1.16
+    np_random_generator = np.random.Generator
+    LEGACY_RNG = False
+except AttributeError:
+
+    class np_random_generator:
+        pass
+
+    LEGACY_RNG = True
+
 logger = logging.getLogger(__name__)
 
 
+class _BackwardsCompatibleNumpyRng:
+    """Thin wrapper to ensure backwards compatibility between
+    new and old numpy randomness generators.
+    """
+    _rng = None
+
+    def __init__(
+            self,
+            generator_or_seed: Optional[Union[
+                "np_random_generator", np.random.RandomState, int]] = None):
+        if generator_or_seed is None:
+            self._rng = np.random
+        elif isinstance(generator_or_seed, np.random.RandomState):
+            self._rng = generator_or_seed
+        elif isinstance(generator_or_seed, np_random_generator):
+            self._rng = generator_or_seed
+        elif LEGACY_RNG:
+            self._rng = np.random.RandomState(generator_or_seed)
+        else:
+            self._rng = np.random.default_rng(generator_or_seed)
+
+    @property
+    def legacy_rng(self) -> bool:
+        return not isinstance(self._rng, np_random_generator)
+
+    @property
+    def rng(self):
+        return self._rng if self._rng else np.random
+
+    def __getattr__(self, name: str) -> Any:
+        # https://numpy.org/doc/stable/reference/random/new-or-different.html
+        if self.legacy_rng:
+            if name == "integers":
+                name = "randint"
+            elif name == "random":
+                name = "rand"
+        return getattr(self.rng, name)
+
+
+RANDOM_STATE = Union[None, _BackwardsCompatibleNumpyRng, np_random_generator,
+                     np.random.RandomState, int]
```

Comment:
Let's call this `RandomState` since it represents a type and not a global variable
```suggestion
RandomState = Union[None, _BackwardsCompatibleNumpyRng, np_random_generator,
                    np.random.RandomState, int]
```